bug: invalid memory address or nil pointer dereference at getFileSHA()

## Describe the Bug

invalid memory address or nil pointer dereference at getFileSHA()

```
2021/12/28 13:18:02 language is go-1.17
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x12bbfc7e0]

goroutine 1 [running]:
github.com/merico-dev/stream/internal/pkg/githubactions.(*GithubActions).getFileSHA(0x1400006a7e0, {0x12bc03c3c, 0xe})
	github.com/merico-dev/stream/internal/pkg/githubactions/githubactions.go:125 +0x1a0
github.com/merico-dev/stream/internal/pkg/githubactions.(*GithubActions).DeleteWorkflow(0x1400006a7e0, 0x140001c65d0)
	github.com/merico-dev/stream/internal/pkg/githubactions/githubactions.go:80 +0x60
github.com/merico-dev/stream/internal/pkg/githubactions.Uninstall(0x14000113c10)
	github.com/merico-dev/stream/internal/pkg/githubactions/uninstall.go:17 +0x218
github.com/merico-dev/stream/cmd/githubactions.Plugin.Uninstall({0x0, 0x0}, 0x14000113c10)
	github.com/merico-dev/stream/cmd/githubactions/main.go:27 +0x34
github.com/merico-dev/stream/internal/pkg/pluginengine.Uninstall(0x14000113bf0)
	github.com/merico-dev/stream/internal/pkg/pluginengine/plugin.go:42 +0xd4
github.com/merico-dev/stream/internal/pkg/planmanager.(*Plan).Execute(0x14000113b60)
	github.com/merico-dev/stream/internal/pkg/planmanager/plan.go:61 +0x3b0
main.installCMDFunc(0x10348b980, {0x1400006a3e0, 0x0, 0x2})
	github.com/merico-dev/stream/cmd/devstream/install.go:45 +0x318
github.com/spf13/cobra.(*Command).execute(0x10348b980, {0x1400006a3c0, 0x2, 0x2})
	github.com/spf13/cobra@v1.2.1/command.go:860 +0xa04
github.com/spf13/cobra.(*Command).ExecuteC(0x10348bc00)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x7b8
github.com/spf13/cobra.(*Command).Execute(0x10348bc00)
	github.com/spf13/cobra@v1.2.1/command.go:902 +0x30
main.main()
	github.com/merico-dev/stream/cmd/devstream/main.go:28 +0x30
```

![image](https://user-images.githubusercontent.com/18692628/147531364-69c933fe-baab-4753-97f1-7e049d4b320a.png)


## Other notes

I've fixed this bug at #74 but it's reopen at #76 by "resolve conflict". 

![image](https://user-images.githubusercontent.com/18692628/147531231-a19d611e-6927-4cb3-99c5-2fc01f731162.png)

We should pay more attentions at "avoid confict"/"resolve conflict"/"code review"

/kind bug
/cc @merico-dev/devstream to us all.

--- a/internal/pkg/githubactions/githubactions.go
+++ b/internal/pkg/githubactions/githubactions.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"strings"
 
 	"github.com/google/go-github/v40/github"
 	"github.com/mitchellh/mapstructure"
@@ -109,10 +110,10 @@ func (ga *GithubActions) DeleteWorkflow(workflow *Workflow) error {
 	return nil
 }
 
-// getFileSHA: 
-// 1. if file exists without error: return (string(SHA), nil)
-// 2. if some errors occurred: return ("", err)
-// 3. if file not found without error: return ("", nil)
+// getFileSHA will try to collect the SHA hash value of the file, then return it. the return values will be:
+// 1. If file exists without error -> string(SHA), nil
+// 2. If some errors occurred -> return "", err
+// 3. If file not found without error -> return "", nil
 func (ga *GithubActions) getFileSHA(filename string) (string, error) {
 	content, _, resp, err := ga.client.Repositories.GetContents(
 		ga.ctx,
@@ -122,14 +123,17 @@ func (ga *GithubActions) getFileSHA(filename string) (string, error) {
 		&github.RepositoryContentGetOptions{},
 	)
 
+	// error reason is not 404
+	if err != nil && !strings.Contains(err.Error(), "404") {
+		return "", err
+	}
+
+	// error reason is 404
 	if resp.StatusCode == http.StatusNotFound {
 		return "", nil
 	}
 
-	if err != nil {
-		return "", err
-	}
-
+	// no error occurred
 	if resp.StatusCode == http.StatusOK {
 		return *content.SHA, nil
 	}

